python/pyspark/sql/dataframe.py

```diff
@@ -1881,6 +1881,15 @@ def toDF(self, *cols):
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
 
+    @since(2.3)
+    def colRegex(self, colName):
+        """
+        Selects column based on the column name specified as a regex and return it
+        as :class:`Column`.
+        """
+        jc = self._jdf.colRegex(colName)
```

Comment:
Could we add a type check here too?

Reply:
@HyukjinKwon Thank you very much for your comments. I will submit changes soon.